Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -876,19 +873,11 @@ public static void enableCookieBasedMatching(String cookieMatchDomain, int delay
      */
     private void executeClose() {
         if (initState_ != SESSION_STATE.UNINITIALISED) {
-            if (!hasNetwork_) {
```

Comment:
The `if` block can only happen when init session request is in flight, in which case the removing it from the queue has no effect, the request will complete either way + it will result in a bug because the `else` block won't be executed which would clean up the last session params (the way this owuld manifest is that during the next session, if user queries last session params before actually initializing the session, they will get valid values instead of "BNC_NO_VALUE".)